downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1432 @@
+
+##### GraphQL Module Improvements
+
+###### Introspection Support
+
+Ballerina GraphQL services now support introspection queries on the schema.
+
+###### Improved Resource Functions
+
+GraphQL resources can now return values union with  `error` values. 
+
+##### Common Changes in Messaging Modules
+
+- The `resource` functions are changed to `remote` functions in the new listener APIs. 
+- The `service` name is given as a string with the new Ballerina language changes.
+
+##### Kafka Module Improvements
+
+###### Client Changes
+
+- The `kafka:Consumer` is separated into `kafka:Listener` (asynchronous) and `kafka:Consumer` (synchronous). 
+- The return type of the `init` functions of the `kafka:Producer` and `kafka:Consumer` is changed to `Error?`. 
+- The `subscribeToPattern()` of the `kafka:Consumer` is changed to `subscribeWithPattern()`.
+- A new record type named `ProducerRecord` is introduced for sending messages. 
+
+    ```ballerina
+    public type ProducerRecord record {|
+        string topic;
+        byte[] key?;
+        byte[] value;
+        int timestamp?;
+        int partition?;
+    |};
+    ```
+
+- The `send()` method of the `kafka:Producer` is changed to `sendProducerRecord(ProducerRecord producerRecord)`. An example of sending a message with the new API is given below.
+
+    ```ballerina
+    string message = "Hello World, Ballerina";
+    kafkaProducer->sendProducerRecord({ topic: "test-kafka-topic",
+                                        value: message.toBytes() });
+    ```
+
+###### Service and Listener Changes
+
+- The return type of the `kafka:Listener init` is changed to `Error?`. 
+- The `resource` functions are changed to `remote` functions in the new listener APIs.
+- Has a single type of service that supports the two types of remote functions below:
+    - `onConsumerRecord(kafka:ConsumerRecord[] record) {}` 
+    - `onConsumerRecord(kafka:Caller caller, kafka:ConsumerRecord[] record) {}`
+- The `kafka:Caller` is introduced to remote functions to commit offsets of the consumed records.
+- The new syntax with the service changes is given below.
+
+    ```ballerina
+    listener kafka:Listener kafkaListener = new (consumerConfigs);
+
+    service kafka:Service on kafkaListener {
+    remote function onConsumerRecord(kafka:Caller caller, kafka:ConsumerRecord[] records) {
+        // Process consumed records
+        }
+    }
+    ```
+
+##### NATS Module Improvements
+
+ - The `ballerinax/nats` module is split into two packages as `ballerinax/nats` (NATS client) and `ballerinax/stan` (NATS Streaming client).
+- The `nats:Connection` is removed entirely. 
+- The `nats:Producer` is renamed to `nats:Client`. 
+- The return type of the `nats:Client` is changed to `Error?`.
+- The `nats:Message` object is changed into a record type. 
+
+    ```ballerina
+    public type Message record {|
+    byte[] content;
+    string subject;
+    string replyTo?;
+    |};
+    ```
```

Comment:
```suggestion
    ```ballerina
    public type Message record {|
           byte[] content;
           string subject;
           string replyTo?;
    |};
    ```
```
Please indent the fields of this record.